akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpCookie.scala

```diff
@@ -140,3 +259,29 @@ object HttpCookie {
   private[http] val domainChars = ALPHANUM ++ ".-"
   private[http] val pathOrExtChars = VCHAR ++ ' ' -- ';'
 }
+
+sealed trait SameSite extends Renderable {
+  def asJava: jm.headers.SameSite = this match {
+    case SameSite.Strict => jm.headers.SameSite.Strict
+    case SameSite.Lax    => jm.headers.SameSite.Lax
+    case SameSite.None   => jm.headers.SameSite.None
+  }
+
+  override private[http] def render[R <: Rendering](r: R): r.type = r ~~ (this match {
+    case SameSite.Strict => "Strict"
+    case SameSite.Lax    => "Lax"
+    case SameSite.None   => "None"
+  })
+}
+
+object SameSite {
+  def apply(s: String): Option[SameSite] = {
+    if ("Lax".equalsIgnoreCase(s)) Some(Lax)
+    else if ("Strict".equalsIgnoreCase(s)) Some(Strict)
+    else if ("None".equalsIgnoreCase(s)) Some(None)
+    else Option.empty
+  }
+  case object Strict extends SameSite
+  case object Lax extends SameSite
+  case object None extends SameSite
```

Comment:
Should we document here why `None` is useful and different from not having a SameSite parameter at all?